### PR TITLE
Add unit tests for models

### PR DESCRIPTION
**Pull request by singh1114**
* Add unit tests for models
* Change models according to the unit tests
* Remove `CeleryScan` model class

Signed-off-by: Ranvir Singh ranvir.singh1114@gmail.com

### CODE REVIEW

File: scanapp/models.py [hunk 1]
```diff
@@ -50,13 +50,19 @@ def __str__(self):
 
     scan_type = models.CharField(max_length=20, choices=scan_types, default='URL')
     is_complete = models.BooleanField()
+    
+    class Meta:
+        verbose_name_plural = 'Scan Info'
```
Comment (pombredanne) [inline scanapp/models.py]: This meta is not needed IMHO.

File: scanapp/models.py [hunk 2]
```diff
@@ -50,13 +50,19 @@ def __str__(self):
...
 
 class UserInfo(models.Model):
     def __str__(self):
         return self.user.username
 
     user = models.OneToOneField(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
     scan_info = models.ForeignKey(ScanInfo)
+    
+    class Meta:
+        verbose_name_plural = 'User Info'
```
Comment (pombredanne) [inline scanapp/models.py]: This meta is not needed IMHO.

File: scanapp/models.py [hunk 3]
```diff
@@ -66,29 +72,37 @@ def __str__(self):
     scan_info = models.ForeignKey(ScanInfo)
     URL = models.URLField(max_length=2000)
 
+    class Meta:
```
Comment (pombredanne) [inline scanapp/models.py]: This meta is not needed IMHO.

File: scanapp/models.py [hunk 4]
```diff
@@ -66,29 +72,37 @@ def __str__(self):
...
 class LocalScanInfo(models.Model):
     def __str__(self):
         return self.folder_name
 
     scan_info = models.ForeignKey(ScanInfo)
     folder_name = models.CharField(max_length=200)
-
+    
+    class Meta:
+        verbose_name_plural = 'Local Scan Info'
```
Comment (pombredanne) [inline scanapp/models.py]: This meta is not needed IMHO.

File: scanapp/models.py [hunk 5]
```diff
@@ -66,29 +72,37 @@ def __str__(self):
...
 class CodeInfo(models.Model):
     def __str__(self):
-        return self.total_code_files
+        return str(self.total_code_files)
 
     scan_info = models.ForeignKey(ScanInfo)
     total_code_files = models.IntegerField(null=True, blank=True)
     code_size = models.IntegerField(null=True, blank=True, default=0)
 
+    class Meta:
```
Comment (pombredanne) [inline scanapp/models.py]: This meta is not needed IMHO.

File: scanapp/models.py [hunk 6]
```diff
@@ -107,7 +121,9 @@ def __str__(self):
 
     scan_result = models.ForeignKey(ScanResult)
     file_path = models.CharField(max_length=400)
-
+    
+    class Meta:
+        verbose_name_plural = 'Scan File Info'
```
Comment (pombredanne) [inline scanapp/models.py]: This meta is not needed IMHO. Also why do you suffix all your models with `Info`?if these are objects they represent something not SomethingInfo

File: scanapp/models.py [hunk 7]
```diff
@@ -127,31 +143,41 @@ def __str__(self):
     owner = models.CharField(max_length=500)
     dejacode_url = models.URLField(max_length=2000)
 
+    class Meta:
+        verbose_name_plural = 'Licenses'
```
Comment (pombredanne) [inline scanapp/models.py]: This meta is not needed IMHO.

File: scanapp/models.py [hunk 8]
```diff
@@ -127,31 +143,41 @@ def __str__(self):
...
-        return self.license_choice
+        return str(self.license_choice)
 
     license = models.ForeignKey(License)
     license_choice = models.BooleanField()
     identifier = models.CharField(max_length=200)
-
+    
+    class Meta:
+        verbose_name_plural = 'Matched Rule'
```
Comment (pombredanne) [inline scanapp/models.py]: This meta is not needed IMHO. And this is not plural

File: scanapp/tests/test_tasks.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
+from django.test import TestCase
+from django.core.files import File
+import json
```
Comment (pombredanne) [inline scanapp/tests/test_tasks.py]: Typically the `json` import which is stdlib should come before django imports that are from a third-party

File: scanapp/tests/test_tasks.py [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+from scanapp.models import ScanFileInfo
+from scanapp.models import License
+from scanapp.models import MatchedRule
+from scanapp.models import MatchedRuleLicenses
+
+class SaveResultsToDBTestCase(TestCase):
+    def test_correct_save_result_to_db_with_scan_type_url(self):
+        # TODO function not complete add more stuff to scanapp/tests/sample_json.txt
+        # to test more
+        URL='https://github.com/'
```
Comment (pombredanne) [inline scanapp/tests/test_tasks.py]: The spacing around equal is not right.

**Review by pombredanne — CHANGES_REQUESTED** [review]
This starts looking good. We need to have a good discussion on models. There are too many of them IMHO. Also do not add meta unless really needed. 
Finally please check vulnerable code and enable the code styles checks ... I have activated Travis on the repo now... so https://travis-ci.org/nexB/scancode-server will show the test runs.

File: scanapp/tests/test_tasks.py (hunk 10 above)
Comment (singh1114) [reply scanapp/tests/test_tasks.py]: OK.

**Review by pombredanne — COMMENTED** [review]
Looking good at a high level. I did not yet review the tests in details...
Now avoid to keep commented out code... Either you have the code and the test pass or fails or you do not keep the code as comment... I guess here this is just because you still have some things to refine on the testing side once merged?

**Pull request merged**